### ISSUE TITLE
Unify PBC to be a torch.tensor of bools

### ISSUE THREAD

**Issue by curtischong** [Radical-AI/torch-sim#251]
Right now we do not have a standardized method to represent booleans. we use `pbc=True` and `pbc=torch.Tensor([True,True,True])` elsewhere. I propose to change it all to `torch.Tensor([True,True,True])`. This is slightly breaking but I think it's better than `pbc=True` because it brings us onto a path to implement axis-specific periodic boundaries.

**Comment by orionarcher** [comment]
I am definitely supportive of this refactor.

**Comment by curtischong** [comment]
I just looked at this and it's slightly painful. This is because Vesin doesn't support mixed PBC: https://github.com/Luthaf/vesin/issues/54

Like the previous commenters suggested, we could write code that creates a vacuum per axis before we do the PBC (but this isn't ideal since it's more compute in python not c). I might try submitting a PR to vesin

**Comment by curtischong** [comment]
we'll see if this PR makes it into vesin https://github.com/Luthaf/vesin/pull/75

**Comment by curtischong** [comment]
done https://github.com/TorchSim/torch-sim/pull/320
